torch/nn/functional.py

```diff
@@ -1130,10 +1134,13 @@ def poisson_nll_loss(input, target, log_input=True, full=False, size_average=Tru
     if full:
         mask = target > 1
         loss[mask] += (target * torch.log(target) - target + 0.5 * torch.log(2 * math.pi * target))[mask]
-    if size_average:
-        return torch.mean(loss)
+    if not reduce:
+      return loss
     else:
-        return torch.sum(loss)
+      if size_average:
+          return torch.mean(loss)
+      else:
+          return torch.sum(loss)
```

Comment:
nit: It's a little cleaner if you get rid of the elses, ie:
```
if not reduce:
    return loss
if size_average:
    ...
```
but this might just be my personal preference.